source/system/timer.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * \file timer.cpp
+ * \author Sahak Grigoryan <sahakgrigoryan.am@gmail.com>
+ * \brief Class implementation: TimerSystem
+ */
+
+//--------------------------------------------------------------------------
+// includes
+// own header
+#include "garden/system/timer.hpp"
+
+// C/C++ standard headers
+
+// third-party library headers
+#include "linear-pool.hpp"
+
+// project-specific headers
+
+namespace garden
+{
+    
+TimerSystem::TimerSystem(void) :
+    System{},
+    Singleton<TimerSystem>{},
+    handlerPool_{},
+    activeHandlers_{},
+    inactiveHandlers_{}
+{
+    // ensure: (lowest supported time ratio) <= millisecond
+    static_assert(std::ratio_less_equal_v<ClockType::period, std::milli>);
+    ECSM_SUBSCRIBE_TO_EVENT("Timer", TimerSystem::runTimers);
+    ECSM_SUBSCRIBE_TO_EVENT("Timer", TimerSystem::disposeTimers);
```

Comment:
Please see the [ComponentsSystem<>](https://github.com/cfnptr/ecsm/blob/2c7a01d2ba7f18121e8c7b584f6b45f004677670/include/ecsm.hpp#L1140) ```disposeComponents()``` function for reference.